src/services/machines/siopV2MachineService.ts

```diff
@@ -0,0 +1,158 @@
+import {v4 as uuidv4} from 'uuid';
+import {VerifiedAuthorizationRequest} from '@sphereon/did-auth-siop';
+import {
+  ConnectionTypeEnum,
+  CorrelationIdentifierEnum,
+  IBasicIdentity,
+  IContact,
+  IdentityRoleEnum,
+  IDidAuthConfig,
+} from '@sphereon/ssi-sdk.data-store';
+import {siopGetRequest, siopSendAuthorizationResponse} from '../../providers/authentication/SIOPv2Provider';
+import {SiopV2AuthorizationRequestData, SiopV2MachineContext} from '../../types/machines/siopV2';
+import {URL} from 'react-native-url-polyfill';
+import {getContacts} from '../contactService';
+import store from '../../store';
+import {addIdentity} from '../../store/actions/contact.actions';
+import {W3CVerifiableCredential} from '@sphereon/ssi-types';
+import {IIdentifier} from '@veramo/core';
+import {getOrCreatePrimaryIdentifier} from '../identityService';
+import {translateCorrelationIdToName} from '../../utils/CredentialUtils';
+
+export const createConfig = async (context: Pick<SiopV2MachineContext, 'requestData' | 'identifier'>): Promise<IDidAuthConfig> => {
+  const {requestData} = context;
+
+  if (requestData?.uri === undefined) {
+    return Promise.reject(Error('Missing request uri in context'));
+  }
+
+  const identifier: IIdentifier = await getOrCreatePrimaryIdentifier();
+  return {
+    id: uuidv4(),
+    // FIXME: Update these values in SSI-SDK. Only the URI (not a redirectURI) would be available at this point
+    sessionId: uuidv4(),
+    redirectUrl: requestData.uri,
+    stateId: requestData.state,
+    identifier,
+  };
+};
+
+export const getSiopRequest = async (
+  context: Pick<SiopV2MachineContext, 'didAuthConfig' | 'requestData'>,
+): Promise<SiopV2AuthorizationRequestData> => {
+  const {didAuthConfig, requestData} = context;
+
+  if (requestData?.uri === undefined) {
+    return Promise.reject(Error('Missing request uri in context'));
+  }
+
+  if (didAuthConfig === undefined) {
+    return Promise.reject(Error('Missing config in context'));
+  }
+
+  const verifiedAuthorizationRequest: VerifiedAuthorizationRequest = await siopGetRequest(didAuthConfig);
+  const name = verifiedAuthorizationRequest.registrationMetadataPayload?.client_name;
+  const url =
+    verifiedAuthorizationRequest.responseURI ??
+    (requestData.uri.includes('request_uri')
+      ? decodeURIComponent(requestData.uri.split('?request_uri=')[1].trim())
+      : verifiedAuthorizationRequest.issuer ?? verifiedAuthorizationRequest.registrationMetadataPayload?.client_id);
+  const uri: URL | undefined = url.includes('://') ? new URL(url) : undefined;
+  const correlationIdName = uri
+    ? translateCorrelationIdToName(uri.hostname)
+    : verifiedAuthorizationRequest.issuer
+    ? translateCorrelationIdToName(verifiedAuthorizationRequest.issuer.split('://')[1])
+    : name;
+  const correlationId: string = uri?.hostname ?? correlationIdName;
```

Comment:
````
uri?.hostname ??
````
Isn't this already covered in the correlationIdName builder? If not, maybe create a buikdCcorrelationIdName() function to only create it when needed

Reply:
no because that leads to a name. i think i'm going to leave this for now, there are so many things we can improve in the code but i think it's better to do that later at some point